src/generator/generator_visitor.ts

```diff
@@ -103,37 +106,27 @@ class GeneratorVisitor extends Visitor {
     protected visitBinaryExpression(node: BinaryExpression) {
         super.visitBinaryExpression(node);
 
-        let right = this.popExpression();
```

Comment:
This new generic way of handling things introduces new special cases. Maybe there's a better solution to this. We can discuss it in person if you want to.